core/src/main/scala/kanaloa/reactive/dispatcher/queue/Queue.scala

```diff
@@ -202,24 +202,37 @@ object QueueOfIterator {
     iterator:                Iterator[_],
     dispatchHistorySettings: DispatchHistorySettings,
     defaultWorkSettings:     WorkSettings,
+    sendResultsTo:           Option[ActorRef]        = None,
     metricsCollector:        MetricsCollector        = NoOpMetricsCollector
   ): Props =
-    Props(new QueueOfIterator(iterator, dispatchHistorySettings, defaultWorkSettings, metricsCollector))
+    Props(new QueueOfIterator(iterator, dispatchHistorySettings, defaultWorkSettings, sendResultsTo, metricsCollector))
 }
 
 object Queue {
 
   case class RequestWork(requester: ActorRef)
 
-  case class Enqueue(workMessage: Any, replyTo: Option[ActorRef] = None, workSettings: Option[WorkSettings] = None)
-  object Enqueue {
-    def apply(workMessage: Any, replyTo: ActorRef): Enqueue = Enqueue(workMessage, Some(replyTo))
-  }
+  /**
+   * Enqueue a message. If the message is enqueued successfully, a [[kanaloa.reactive.dispatcher.queue.Queue.WorkEnqueued]] is sent to the sender if `sendAcks` is true.
+   * Any results will be sent to the `replyTo` actor.  If the work is rejected, a [[WorkRejected]] is sent to the sender, regardless of the value of `sendAcks`.
+   * @param workMessage The message to enqueue
+   * @param sendAcks Send ack messages.  This does not control [[WorkRejected]] messages, which are sent regardless for backpressure.
+   * @param sendResultsTo Actor which can optionally receive responses from downstream backends.
+   */
+  case class Enqueue(workMessage: Any, sendAcks: Boolean = false, sendResultsTo: Option[ActorRef] = None)
+
   case object WorkEnqueued
   case object Unregistered
 
   case class Unregister(worker: WorkerRef)
 
+  case class EnqueueRejected(workMessage: Any, reason: EnqueueRejected.Reason, sendResultsTo: Option[ActorRef])
+
```

Comment:
Two reasons:
Yes, we don't want to lose info, so that if the sender wants to retry eventually, it can just resend.  I guess I could just have the `Enqueue` message as a member on the `EnqueueRejected` class, probably a better idea now that I think of it.  Should I make that change?

Secondly, for `PushingDispatcher`, it sets `sendResultsTo` as the original sender, which is who, in the case of a rejection, will need to be sent the `WorkRejected` message.